File: src/main/java/com/gmail/borlandlp/minigamesdtools/party/event/PartyChangeLeaderEvent.java
package com.gmail.borlandlp.minigamesdtools.party.event;

import com.gmail.borlandlp.minigamesdtools.party.Party;
import org.bukkit.entity.Player;
import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;

public class PartyChangeLeaderEvent extends Event {
    private Party party;
    private Player oldLeader;
    private Player newLeader;
    private static final HandlerList handlers = new HandlerList();

    public PartyChangeLeaderEvent(Party party, Player oldLeader, Player newLeader) {
        this.party = party;
        this.oldLeader = oldLeader;
        this.newLeader = newLeader;
    }

    public Party getParty() {
        return party;
    }

    public Player getOldLeader() {
        return oldLeader;
    }

    public Player getNewLeader() {
        return newLeader;
    }

    @Override
    public HandlerList getHandlers()
    {
        return handlers;
    }

    public static HandlerList getHandlerList() {
        return handlers;
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/party/event/PartyChangeLeaderEvent.kt
package com.gmail.borlandlp.minigamesdtools.party.event

import com.gmail.borlandlp.minigamesdtools.party.Party
import org.bukkit.entity.Player
import org.bukkit.event.Event
import org.bukkit.event.HandlerList

class PartyChangeLeaderEvent(val party: Party, val oldLeader: Player, val newLeader: Player) : Event() {

    override fun getHandlers(): HandlerList {
        return handlerList
    }

    companion object {
        val handlerList = HandlerList()
    }

}